src/thin_client.rs

```diff
@@ -91,67 +122,72 @@ impl ThinClient {
         self.transfer_signed(tx).map(|_| sig)
     }
 
-    /// Request the balance of the user holding `pubkey`. This method blocks
-    /// until the server sends a response. If the response packet is dropped
-    /// by the network, this method will hang indefinitely.
+    /// Request the balance of the user holding `pubkey`.
     pub fn get_balance(&mut self, pubkey: &PublicKey) -> io::Result<i64> {
-        trace!("get_balance");
+        info!("get_balance");
         let req = Request::GetBalance { key: *pubkey };
-        let data = serialize(&req).expect("serialize GetBalance in pub fn get_balance");
-        self.requests_socket
-            .send_to(&data, &self.requests_addr)
-            .expect("buffer error in pub fn get_balance");
-        let mut done = false;
-        while !done {
-            let resp = self.recv_response()?;
-            trace!("recv_response {:?}", resp);
-            if let &Response::Balance { ref key, .. } = &resp {
-                done = key == pubkey;
+        let (response_sender, response_receiver) = channel();
+        self.sender.send((Message::Requests(req), response_sender)).expect("send request to thread");
+        let handle = thread::spawn(move || {
+            loop{
+                match response_receiver.try_recv() {
```

Comment:
All these `try_recv()` calls look like they can be `recv()`.

Reply:
Yep, checks out, thanks!